src/ModuleBlacklist/Magerun/SecurityScanCommand.php

```diff
@@ -28,70 +33,199 @@ protected function configure()
-                        $output->writeln(sprintf('<comment>Update URL:</comment> %s', $updateUrl));
-                    }
-                    
-                    if (!empty($credit)) {
-                        $output->writeln(sprintf('<comment>Credit:</comment>     %s', $credit));
-                    }
-                    
-                    $output->writeln('');
-
-                    $exitCode = 1;
+
+            while ($row = $this->getRowObject(fgetcsv($blacklist))) {
+                if ($this->checkIsInstalledModule($output, $row) || $this->checkIsInstalledRoute($output, $row)) {
                     $hitCount++;
                 }
             }
 
             if ($hitCount === 0) {
                 $output->writeln('No known vulnerable modules detected.');
+                return 0;
             }
 
-            return $exitCode;
+            return 1;
+        }
+
+        return 2;
+    }
+
+    /**
+     * Check row for match in installed modules, by version
+     *
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     * @param \Varien_Object $row
+     * @return int
+     */
+    protected function checkIsInstalledModule(OutputInterface $output, \Varien_Object $row)
+    {
+        // No match if module has no installed version, or version is equal/greater than fixed-in.
+        if ($row->getVersion() === null
+            || (!empty($row->getFixedIn()) && version_compare($row->getVersion(), $row->getFixedIn(), '>='))) {
+            return false;
+        }
+
+        $output->writeln(
+            sprintf(
+                '<error>Vulnerable module found: %s%s</error>',
+                $row->getName(),
+                $output->isQuiet() && !empty($row->getFixedIn())
+                    ? sprintf(' (%s < %s)', $row->getVersion(), $row->getFixedIn())
+                    : ''
+            ),
+            OutputInterface::VERBOSITY_QUIET
+        );
+
+        $output->writeln(sprintf('<comment>Installed:</comment>  %s', $row->getVersion()));
+        $output->writeln(sprintf('<comment>Fixed In:</comment>   %s', $row->getFixedIn() ?: '(unknown)'));
+
+        if (!empty($row->getUpdateUrl())) {
+            $output->writeln(sprintf('<comment>Update URL:</comment> %s', $row->getUpdateUrl()));
+        }
+
+        if (!empty($row->getCredit())) {
+            $output->writeln(sprintf('<comment>Credit:</comment>     %s', $row->getCredit()));
+        }
+
+        $output->writeln('');
+
+        return true;
+    }
+
+    /**
+     * Check for match in frontend routes if module is unknown
+     *
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     * @param \Varien_Object $row
+     * @return bool
+     */
+    protected function checkIsInstalledRoute(OutputInterface $output, \Varien_Object $row)
+    {
+        $module = $this->getModuleByRoute($row->getFrontname());
+
+        if ($module === null || ($row->getName() !== '?' && !empty($row->getName()))) {
```

Comment:
OK so this means that when adding a row where the module name is unknown the row **must** start with a question mark or be empty. I think it's worth updating the readme to document this rather than to assume (what if someone enters the string "Unknown"

Reply:
Updated the readme accordingly.